openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+                  'is not one of the scores'.format(score)
+            raise ValueError(msg)
+
+        return score_index
+
+    def get_filter_indices(self, filters=[], filter_bins=[]):
+        """Get indices into the filter axis of this tally's data arrays.
+
+        This is a helper method for the Tally.get_values(...) method to
+        extract tally data. This method returns the indices into the filter
+        axis of the tally's data array (axis=0) for particular combinations
+        of filters and their corresponding bins.
+
+        Parameters
+        ----------
+        filters : Iterable of openmc.FilterMeta
+            An iterable of filter types
+            (e.g., [MeshFilter, EnergyFilter]; default is [])
+        filter_bins : Iterable of tuple
+            A list of tuples of filter bins corresponding to the filter_types
+            parameter (e.g., [(1,), ((0., 0.625e-6),)]; default is []). Each
+            tuple contains bins for the corresponding filter type in the filters
+            parameter. Each bin is an integer ID for Material-, Surface-,
+            Cell-, Cellborn-, and Universe- Filters. Each bin is an integer
+            for the cell instance ID for DistribcellFilters. Each bin is a
+            2-tuple of floats for Energy- and Energyout- Filters corresponding
+            to the energy boundaries of the bin of interest. The bin is an
+            (x,y,z) 3-tuple for MeshFilters corresponding to the mesh cell
+            of interest. The order of the bins in the list must correspond to
+            the filter_types parameter.
+
+        Returns
+        -------
+        numpy.ndarray
+            A NumPy array of the filter indices
+
+        """
+
+        cv.check_type('filters', filters, Iterable, openmc.FilterMeta)
+        cv.check_type('filter_bins', filter_bins, Iterable, tuple)
+
+        # Determine the score indices from any of the requested scores
+        if filters:
+            # Initialize empty list of indices for each bin in each Filter
+            filter_indices = []
+
+            # Loop over all of the Tally's Filters
+            for i, self_filter in enumerate(self.filters):
+                # If a user-requested Filter, get the user-requested bins
+                for j, test_filter in enumerate(filters):
+                    if type(self_filter) is test_filter:
+                        bins = filter_bins[j]
+                        break
+                else:
+                    # If not a user-requested Filter, get all bins
+                    if isinstance(self_filter, openmc.DistribcellFilter):
+                        # Create list of cell instance IDs for distribcell Filters
+                        bins = list(range(self_filter.num_bins))
+
+                    elif isinstance(self_filter, openmc.EnergyFunctionFilter):
+                        # EnergyFunctionFilters don't have bins so just add a None
+                        bins = [None]
```

Comment:
None is still an object in the array, and can get confusing when you're iterating, or adding to it. I think an empty array, or none is more appropriate here.